benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java

```diff
@@ -208,6 +212,13 @@ public static void main(String[] args) throws Exception {
             worker.close();
         }
 
+        // We catch and eat exceptions thrown by benchmark runs above, but we would like the process
+        // to return a non-zero exit code on failure, for the benefit of the caller.
+        if (anyFailed.get()) {
+            System.exit(1);
+        } else {
```

Comment:
Nit: Do we need the else here? If the original conditional isn't met then the main function should just exit normally with a zero exit code.

Reply:
Also I believe this isn't needed as the catch block where `anyFailed` is set to true exits the program with a non-zero code anyhow.

Reply:
We need the else because `System.exit(0)` exits the program even if non-daemon threads are still running which is something that happens in OMB, i.e., it's the big hammer.

You are right that this whole flow changed per the above comment, still evaluating that part.

Reply:
Right, missed that.

Reply:
This simplified, sort of, in the current push.